cdc/sink/common/common.go

```diff
@@ -24,17 +24,43 @@ import (
 	"go.uber.org/zap"
 )
 
+type txnsWithTheSameCommitTs struct {
+	txns     map[model.Ts]*model.SingleTableTxn
+	commitTs model.Ts
+}
+
+func (t *txnsWithTheSameCommitTs) Append(row *model.RowChangedEvent) {
+	if row.CommitTs != t.commitTs {
+		log.Panic("unexpected row change event",
+			zap.Uint64("commitTs of txn", t.commitTs),
+			zap.Any("row", row))
+	}
+	if t.txns == nil {
+		t.txns = make(map[model.Ts]*model.SingleTableTxn)
+	}
+	txn, exist := t.txns[row.StartTs]
+	if !exist {
+		txn = &model.SingleTableTxn{
+			StartTs:  row.StartTs,
+			CommitTs: row.CommitTs,
+			Table:    row.Table,
+		}
+		t.txns[row.StartTs] = txn
+	}
+	txn.Append(row)
+}
+
 // UnresolvedTxnCache caches unresolved txns
 type UnresolvedTxnCache struct {
 	unresolvedTxnsMu sync.Mutex
-	unresolvedTxns   map[model.TableID][]*model.SingleTableTxn
+	unresolvedTxns   map[model.TableID][]txnsWithTheSameCommitTs
```

Comment:
Please add some comments about the design principle of txn cache, such as
- the intput of txn cache is `CommitTs` strictly increasing
- why txn is grouped by `CommitTs`, and then grouped by `StartTs`

Reply:
PTAL again